src/npc/ezloCap.c

```diff
@@ -0,0 +1,358 @@
+extern void gUnk_0801606C; // Cutscene data type?
+
+extern void sub_0809623C(Entity*);
+extern void sub_08096260(Entity*);
+
+const u8 gUnk_08114134[];
+const u8 gUnk_08114144[];
+const Hitbox gUnk_08114154;
+const u8 gUnk_0811415C[];
+const ScreenTransitionData* const gUnk_081141A4[];
+const u8 gUnk_081141E4[];
+const u16 gUnk_081141F4[];
+
+typedef struct {
+    u32 unk_00;
+    s8 unk_04;
+} gUnk_0810C89C_struct;
+
+extern void sub_08078850(Entity*, u32, u8 /* TODO this is a s8 in beedle.c*/, gUnk_0810C89C_struct*);
+
+// Ezlo as a cap in the Minish Woods, is destroyed as soon as link wears ezlo
+void EzloCap(Entity* this) {
+    if (this->action == 0) {
+        this->action += 1;
+        sub_0805E3A0(this, 2);
+        sub_0807DD50(this);
+    }
+    sub_0807DDAC(this, NULL);
+    sub_0807DDE4(this);
+    UpdateAnimationSingleFrame(this);
+}
+
+typedef struct {
+    u8 unk_00[4];
+    u32 field_04;
+    u32 field_08;
+    u8 unk_0c[8];
+    u32 field_14;
+    u8 unk_18;
+    u8 field_19;
+    u8 unk_1a[3];
+    u16 field_1e;
+    u8 unk_20[2];
+    u16 field_22;
+} paramStruct;
+
+void sub_0806D8A0(Entity* this, paramStruct* param_2) {
+    s32 xOffset;
+    s32 yOffset;
+
+    param_2->field_19 = 8;
+    param_2->field_08 |= 2;
+    param_2->field_14 = 0;
+    xOffset = 16;
+    if (this->spriteSettings.b.flipX) {
+        xOffset = -xOffset;
+    }
+    xOffset += gPlayerEntity.x.HALF.HI;
+    yOffset = gPlayerEntity.y.HALF.HI + 2;
+    param_2->field_1e = xOffset;
+    param_2->field_22 = yOffset;
+
+    xOffset -= this->x.HALF.HI;
+    this->direction = sub_080045DA(xOffset, yOffset - this->y.HALF.HI);
+    this->animationState = (this->animationState & 0x80) | gUnk_08114134[this->direction >> 4];
+}
+
+void sub_0806D908(Entity* this) {
+    this->direction = sub_080045D4(this->x.HALF.HI, this->y.HALF.HI, gPlayerEntity.x.HALF.HI, gPlayerEntity.y.HALF.HI);
+    this->animationState = (this->animationState & 0x80) | gUnk_08114144[this->direction >> 0x1];
+}
+
+// called when talk to ezlo, also when ezlo moves after you
+void sub_0806D944(Entity* this) {
+    this->spriteSettings.b.flipX = 0;
+    if (this->x.WORD <= gPlayerEntity.x.WORD) {
+        this->spriteSettings.b.flipX = 1;
+    }
+}
+
+// Ezlo Angry FX
+void sub_0806D96C(Entity* this) {
+    Entity* fx = CreateFx(this, 0x42, 0);
+    if (fx != NULL) {
+        fx->spritePriority.b0 = 1;
+        PositionRelative(this, fx, 0, 0xFFE80000);
+    }
+}
+
+void sub_0806D9A4(Entity* this) {
+    if (this->x.WORD <= gPlayerEntity.x.WORD) {
+        gPlayerEntity.spriteSettings.b.flipX = 0;
+    } else {
+        gPlayerEntity.spriteSettings.b.flipX = 1;
+    }
+    gPlayerEntity.animationState = 4;
+}
+
+// NPC4E is created when link enters minish woods for the first time and
+// destroyed once he wears ezlo. Also exists when entering hyrule for the second time
+// and fusing kinstones. Is destroyed as soon as the kinstone is fused.
+void NPC4E(Entity* this) {
```

Comment:
Perhaps it would make sense to have this in a different TU?

Reply:
Everything else lgtm

Reply:
What is a TU? Just a different c file?
NPC4E seems to be an invisible npc that is also tied to ezlo.
If I put it in a separate file, probably all the function after it also belong to NPC4E and not to EzloCap?

Reply:
Translation Unit (pretty much = c file)

You bring up a good point. Considering how some entities actually share several functions (spear and bow Moblin for example), perhaps it might make sense in some cases. We just haven't done that for anything yet. So I think it should be fine like this.